src/xo-app/vm/tab-disks.js

```diff
@@ -1,146 +1,481 @@
+      <div className='form-group'>
+        <SelectVdi
+          predicate={this._vdiPredicate}
+          containerPredicate={this._srPredicate}
+          onChange={this._selectVdi}
+        />
+      </div>
+      {vdi && <fieldset className='form-inline'>
+        <div className='form-group'>
+          Bootable <Toggle ref='bootable' /> Readonly <Toggle ref='readOnly' />
+        </div>
+        <span className='pull-right'>
+          <ActionButton icon='add' form='attachDiskForm' btnStyle='primary' handler={this._addVdi}>Create</ActionButton>
+        </span>
+      </fieldset>
       }
-    </Col>
-  </Row>
-</Container>
+    </form>
+  }
+}
+
+const orderItemSource = {
+  beginDrag: props => ({
+    id: props.id,
+    index: props.index
+  })
+}
+
+const orderItemTarget = {
+  hover: (props, monitor, component) => {
+    const dragIndex = monitor.getItem().index
+    const hoverIndex = props.index
+
+    if (dragIndex === hoverIndex) {
+      return
+    }
+
+    props.move(dragIndex, hoverIndex)
+    monitor.getItem().index = hoverIndex
+  }
+}
+
+@DropTarget('orderItem', orderItemTarget, connect => ({
+  connectDropTarget: connect.dropTarget()
+}))
+@DragSource('orderItem', orderItemSource, (connect, monitor) => ({
+  connectDragSource: connect.dragSource(),
+  isDragging: monitor.isDragging()
+}))
+@propTypes({
+  connectDragSource: propTypes.func.isRequired,
+  connectDropTarget: propTypes.func.isRequired,
+  index: propTypes.number.isRequired,
+  isDragging: propTypes.bool.isRequired,
+  id: propTypes.any.isRequired,
+  item: propTypes.object.isRequired,
+  move: propTypes.func.isRequired
+})
+class OrderItem extends Component {
+  _toggle = checked => {
+    const { item } = this.props
+    item.active = checked
+    this.forceUpdate()
+  }
+
+  render () {
+    const { item, connectDragSource, connectDropTarget } = this.props
+    return connectDragSource(connectDropTarget(
+      <li className='list-group-item'>
+        <Icon icon='grab' />
+        {' '}
+        <Icon icon='grab' />
+        {' '}
+        {item.text}
+        <span className='pull-right'>
+          <Toggle value={item.active} onChange={this._toggle} />
+        </span>
+      </li>
+    ))
+  }
+}
+
+@propTypes({
+  onClose: propTypes.func,
+  vm: propTypes.object.isRequired
+})
+@DragDropContext(HTML5Backend)
+class BootOrder extends Component {
+  constructor (props) {
+    super(props)
+    const { vm } = props
+    const order = this._parseBootOrder(vm.boot && vm.boot.order)
+    this.state = {order}
+  }
+
+  _parseBootOrder (bootOrder) {
+    const bootOptions = {
+      c: 'Hard-Drive',
+      d: 'DVD-Drive',
+      n: 'Network'
+    }
+    const order = []
+    if (bootOrder) {
+      for (let id of bootOrder) {
+        if (id in bootOptions) {
+          order.push({id, text: bootOptions[id], active: true})
+          delete bootOptions[id]
+        }
+      }
+    }
+    forEach(bootOptions, (text, id) => order.push({id, text, active: false}))
```

Comment:
Be careful of the iteratee returned value in a `forEach`, it works here but it's usually a good idea to put braces.